### PR TITLE
statistics: use histogram buckets bounds to enhance string match functions estimation

### PR DESCRIPTION
**Pull request by time-and-fate**
cherry-pick #40338 to [release-6.5-20230104-v6.5.0](https://github.com/pingcap/tidb/tree/release-6.5-20230104-v6.5.0)

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by qw4990 — DISMISSED** [review]

File: statistics/table.go [hunk 1]
```diff
@@ -685,9 +686,7 @@ func (coll *HistColl) findAvailableStatsForCol(sctx sessionctx.Context, uniqueID
 // GetSelectivityByFilter try to estimate selectivity of expressions by evaluate the expressions using TopN and NULL.
```
Comment (chrysan) [inline statistics/table.go:686]: Should this comment be updated?

Comment (time-and-fate) [reply statistics/table.go]: Updated.

**Review by chrysan — APPROVED** [review]

**Pull request merged**